### ISSUE TITLE
Errors in Alfresco 5.0.2.1

### ISSUE THREAD

**Issue by bsreid** [wabson/share-import-export#13]
Trying to use these scripts to copy some sites from a 4.2.2 environment into a 5.0.2.1 environment.  The exports from the 4.2.2 environment worked great. But, now I'm having problems getting the data into my 5.0.2.1 site.

When I just try to ping the server using the ping.py script, I get the following errors:

C:\Windows\System32>python C:\Users\sreid\Documents\Projects\share-import-export-master\shareclient\ping.py -u admin -p admin -U http://127.0.0.1:8080/share
Log in (admin)
Log out (admin)
Traceback (most recent call last):
  File "C:\Users\sreid\Documents\Projects\share-import-export-master\shareclient\ping.py", line 94, in <module>
    main(sys.argv[1:])
  File "C:\Users\sreid\Documents\Projects\share-import-export-master\shareclient\ping.py", line 91, in main
    sc.doLogout()
  File "C:\Users\sreid\Documents\Projects\share-import-export-master\shareclient\alfresco.py", line 353, in doLogout
    resp = self.doPost('page/dologout')
  File "C:\Users\sreid\Documents\Projects\share-import-export-master\shareclient\alfresco.py", line 275, in doPost
    return self.doRequest(method, path, data, dataType)
  File "C:\Users\sreid\Documents\Projects\share-import-export-master\shareclient\alfresco.py", line 263, in doRequest
    raise SurfRequestError(method, e.url, e.code, e.msg, e.hdrs, e.fp)
alfresco.SurfRequestError

And, running it with the debug option looks like this:
C:\Windows\System32>python C:\Users\sreid\Documents\Projects\share-import-export-master\shareclient\ping.py -u admin -p admin -U http://localhost:8080/share -d
Log in (admin)
POST http://localhost:8080/share/login
send: 'POST /share/login HTTP/1.1\r\nAccept-Encoding: identity\r\nContent-Length: 120\r\nAccept-Language: en-gb,en;q=0.5\r\nConnection: clos
e\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nUser-Agent: ShareImportExport/1.0\r\nAccept-Charset: ISO-885
9-1,utf-8;q=0.7,*;q=0.7\r\nHost: localhost:8080\r\nContent-Type: application/x-www-form-urlencoded\r\n\r\nusername=admin&failure=%2Fshare%2F
page%2Ftype%2Flogin%3Ferror%3Dtrue&password=admin&success=%2Fshare%2Fpage%2Fsite-index'
reply: 'HTTP/1.1 404 Not Found\r\n'
header: Server: Apache-Coyote/1.1
header: Content-Type: text/html;charset=utf-8
header: Content-Language: en
header: Content-Length: 973
header: Date: Tue, 13 Oct 2015 16:11:40 GMT
header: Connection: close
POST http://localhost:8080/share/page/dologin
send: 'POST /share/page/dologin HTTP/1.1\r\nAccept-Encoding: identity\r\nContent-Length: 120\r\nAccept-Language: en-gb,en;q=0.5\r\nConnectio
n: close\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nUser-Agent: ShareImportExport/1.0\r\nAccept-Charset:
ISO-8859-1,utf-8;q=0.7,*;q=0.7\r\nHost: localhost:8080\r\nContent-Type: application/x-www-form-urlencoded\r\n\r\nusername=admin&failure=%2Fs
hare%2Fpage%2Ftype%2Flogin%3Ferror%3Dtrue&password=admin&success=%2Fshare%2Fpage%2Fsite-index'
reply: 'HTTP/1.1 302 Found\r\n'
header: Server: Apache-Coyote/1.1
header: X-Frame-Options: SAMEORIGIN
header: X-Content-Type-Options: nosniff
header: X-XSS-Protection: 1; mode=block
header: Set-Cookie: JSESSIONID=999CEBD16949FB2B5B8601C8C0D228A6; Path=/share/; HttpOnly
header: Set-Cookie: alfLogin=1444752701; Expires=Tue, 20-Oct-2015 16:11:41 GMT; Path=/share
header: Set-Cookie: alfUsername3=admin; Expires=Tue, 20-Oct-2015 16:11:41 GMT; Path=/share
header: Location: http://localhost:8080/share/page/site-index
header: Content-Length: 0
header: Date: Tue, 13 Oct 2015 16:11:41 GMT
header: Connection: close
send: 'GET /share/page/site-index HTTP/1.1\r\nAccept-Encoding: identity\r\nAccept-Language: en-gb,en;q=0.5\r\nConnection: close\r\nAccept: t
ext/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nUser-Agent: ShareImportExport/1.0\r\nAccept-Charset: ISO-8859-1,utf-8;q=0.
7,*;q=0.7\r\nHost: localhost:8080\r\nCookie: JSESSIONID=999CEBD16949FB2B5B8601C8C0D228A6; alfLogin=1444752701; alfUsername3=admin\r\n\r\n'
reply: 'HTTP/1.1 302 Found\r\n'
header: Server: Apache-Coyote/1.1
header: X-Frame-Options: SAMEORIGIN
header: X-Content-Type-Options: nosniff
header: X-XSS-Protection: 1; mode=block
header: Cache-Control: no-cache
header: Location: http://localhost:8080/share/page/user/admin/dashboard
header: Content-Type: text/html;charset=utf-8
header: Content-Language: en-GB
header: Content-Length: 0
header: Date: Tue, 13 Oct 2015 16:11:42 GMT
header: Connection: close
send: 'GET /share/page/user/admin/dashboard HTTP/1.1\r\nAccept-Encoding: identity\r\nAccept-Language: en-gb,en;q=0.5\r\nConnection: close\r\
nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nUser-Agent: ShareImportExport/1.0\r\nAccept-Charset: ISO-8859-1,
utf-8;q=0.7,*;q=0.7\r\nHost: localhost:8080\r\nCookie: JSESSIONID=999CEBD16949FB2B5B8601C8C0D228A6; alfLogin=1444752701; alfUsername3=admin\
r\n\r\n'
reply: 'HTTP/1.1 200 OK\r\n'
header: Server: Apache-Coyote/1.1
header: X-Frame-Options: SAMEORIGIN
header: X-Content-Type-Options: nosniff
header: X-XSS-Protection: 1; mode=block
header: Cache-Control: no-cache
header: Content-Type: text/html;charset=utf-8
header: Content-Language: en-GB
header: Transfer-Encoding: chunked
header: Date: Tue, 13 Oct 2015 16:11:43 GMT
header: Connection: close
Log out (admin)
GET http://localhost:8080/share/page/dologout
send: 'GET /share/page/dologout HTTP/1.1\r\nAccept-Encoding: identity\r\nAccept-Language: en-gb,en;q=0.5\r\nConnection: close\r\nAccept: tex
t/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nUser-Agent: ShareImportExport/1.0\r\nAccept-Charset: ISO-8859-1,utf-8;q=0.7,
*;q=0.7\r\nHost: localhost:8080\r\nCookie: JSESSIONID=999CEBD16949FB2B5B8601C8C0D228A6; alfLogin=1444752701; alfUsername3=admin\r\n\r\n'
reply: 'HTTP/1.1 405 Method Not Allowed\r\n'
header: Server: Apache-Coyote/1.1
header: X-Frame-Options: SAMEORIGIN
header: X-Content-Type-Options: nosniff
header: X-XSS-Protection: 1; mode=block
header: Allow: HEAD, POST, OPTIONS
header: Content-Type: text/html;charset=utf-8
header: Content-Language: en
header: Content-Length: 1045
header: Date: Tue, 13 Oct 2015 16:11:44 GMT
header: Connection: close
POST http://localhost:8080/share/page/dologout
send: 'POST /share/page/dologout HTTP/1.1\r\nAccept-Encoding: identity\r\nContent-Length: 0\r\nAccept-Language: en-gb,en;q=0.5\r\nHost: loca
lhost:8080\r\nAccept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8\r\nUser-Agent: ShareImportExport/1.0\r\nAccept-Charset
: ISO-8859-1,utf-8;q=0.7,*;q=0.7\r\nConnection: close\r\nCookie: JSESSIONID=999CEBD16949FB2B5B8601C8C0D228A6; alfLogin=1444752701; alfUserna
me3=admin\r\nContent-Type: application/x-www-form-urlencoded\r\n\r\n'
reply: 'HTTP/1.1 401 Unauthorized\r\n'
header: Server: Apache-Coyote/1.1
header: X-Frame-Options: SAMEORIGIN
header: X-Content-Type-Options: nosniff
header: X-XSS-Protection: 1; mode=block
header: Set-Cookie: alfUsername3=""; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Path=/share
header: Location: /share
header: Content-Length: 0
header: Date: Tue, 13 Oct 2015 16:11:45 GMT
header: Connection: close
Traceback (most recent call last):
  File "C:\Users\sreid\Documents\Projects\share-import-export-master\shareclient\ping.py", line 94, in <module>
    main(sys.argv[1:])
  File "C:\Users\sreid\Documents\Projects\share-import-export-master\shareclient\ping.py", line 91, in main
    sc.doLogout()
  File "C:\Users\sreid\Documents\Projects\share-import-export-master\shareclient\alfresco.py", line 353, in doLogout
    resp = self.doPost('page/dologout')
  File "C:\Users\sreid\Documents\Projects\share-import-export-master\shareclient\alfresco.py", line 275, in doPost
    return self.doRequest(method, path, data, dataType)
  File "C:\Users\sreid\Documents\Projects\share-import-export-master\shareclient\alfresco.py", line 263, in doRequest
    raise SurfRequestError(method, e.url, e.code, e.msg, e.hdrs, e.fp)
alfresco.SurfRequestError

Does anyone have any recommendations for how to get past this issue?  I already disabled the CSRF filter, so that shouldn't be an issue.

**Comment by bsreid** [comment]
Looking at this now, I think that using this tool to move data from 4.x to 5.x may not be a good idea.  However, I'm still concerned about the fact that just using the ping.py script is causing errors to be returned when running against a 5.x environment.  Any guidance on how to fix that problem would be appreciated.

**Comment by wabson** [comment]
The 'ping.py` script should now work OK. I would recommend reading the disclaimer in the README file if you are considering the scripts this to migrate a system - you would be much better doing an in-place upgrade of the system.

**Comment by bsreid** [comment]
Thanks Will.  I appreciate you continuing to update these scripts.  My coworker and I have used them for several years now, and they continue to come in very handy for us.

And, thanks for the reminder about the disclaimer.  This wasn't meant to be a real upgrade; I don't need all of the sites on the old 4.x system, so I was trying some tests to see if I could just get the few sites that I needed.  I may end up doing an in-place upgrade anyway, but it's good to know that the 401 issue has been resolved.

Sean
